src/lib/atemUtil.ts

```diff
@@ -24,6 +26,22 @@ export namespace Util {
 		return value
 	}
 
+	export function subprocessSendPromise (subprocess: ChildProcess | null, message: any) {
+		return new Promise((resolve, reject) => {
+			if (!subprocess) {
+				return resolve()
+			}
+
+			subprocess.send(message, (error: Error) => {
```

Comment:
The promise resolval here is based on not receiving any errors. I don't think it's good enough to resolve for example a `Atem.connect()` call based on the fact that the child process _appears_ to have seen the request. I think we need to send a message id along, store the promise and resolve it when a succes message comes back from the worker thread. (Or perhaps use some library that already implements this)

Reply:
Agreed, I can work o nthat.